okhttp/src/main/java/okhttp3/internal/connection/RealConnection.java

```diff
@@ -276,6 +284,9 @@ private void connectTls(ConnectionSpecSelector connectionSpecSelector) throws IO
             + "\n    subjectAltNames: " + OkHostnameVerifier.allSubjectAltNames(cert));
       }
 
+      X509Certificate cert = (X509Certificate) unverifiedHandshake.peerCertificates().get(0);
+      subjectAlternativeNames = OkHostnameVerifier.allSubjectAltNames(cert);
```

Comment:
... again, not necessarily for this PR. But add a `TODO` comment? Seems wasteful to parse 2x. 

(calling directly through the hostname verifier is annoying because HostnameVerifier is a public type)